R/asserts.R

```diff
@@ -148,6 +148,52 @@ assertInteger <- function(..., type, call = sys.call(-1)) {
     }
 }
 
+#' Check for Factor
+#'
+#' @description Checks if all passed vars and their content are factors or can
+#' be coerced to factor without loss of information.
+#' @param ... Any number of variables as named elements e.g. var1 = var1.
+#' @noRd
+assertFactor <- function(..., type, call = sys.call(-1)) {
```

Comment:
We will also need tests for these new assert functions

Reply:
I didn't end up using it so I just commented it out, I didn't want to delete it outright but let me know if I should store it somewhere besides right inside the code

Reply:
Was there a particular reason you didn't implement the general assert functions. I think we should go ahead and implement them.

Reply:
Yeah - I think it would be worthwhile - as it will possibly avoid duplication of code down the road (either for new functionality or for cleaning up existing code).

Reply:
I uncommented assertFactor and added a test_that. I'm having trouble coming up with a case that would fail the assert because as I understand it the only thing that stops something from being coerced into a factor is if there is an infinite amount of options in the vector, dt, etc that it's in. Unless assertFactor changes to be just whether a value is a factor, and not whether it can be coerced into one, I'm not sure that it would ever return an error.

Reply:
Looks good now.